src/c_api/c_api.cc

```diff
@@ -891,23 +895,51 @@ XGB_DLL int XGBoosterPredict(BoosterHandle handle,
 XGB_DLL int XGBoosterLoadModel(BoosterHandle handle, const char* fname) {
   API_BEGIN();
   CHECK_HANDLE();
-  std::unique_ptr<dmlc::Stream> fi(dmlc::Stream::Create(fname, "r"));
```

Comment:
I am wandering whether this stream supports peek() method. If so we may peak at the header and see whether it is binary format or not?

I am a bit fuzzy on details how binary serialization works now. Do we have some kind of known header?

Reply:
> I am a bit fuzzy on details how binary serialization works now. Do we have some kind of known header?

Not really.  I want to deprecate the binary format, it might still take a long time to actually do it.  Anyway I myself consider it as a burden and don't want to deal with it anymore. ;-)

Reply:
I see. Good to know that native format will become a thing from the past eventually